src/containers/BooksForm.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { createBook } from '../actions';
+
+const categories = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
+
+class BookForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { title: '', category: 'Action' };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.selectHandleChange = this.selectHandleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    const { props: { createBook }, state: { title, category } } = this;
+    const book = {
+      id: Math.random().toString(),
+      title,
+      category,
+    };
+    createBook(book);
+    this.setState({ title: '' });
+  }
+
+  handleChange(event) {
+    this.setState({ title: event.target.value });
+  }
+
+  selectHandleChange(event) {
+    this.setState({ category: event.target.value });
+  }
+
```

Comment:
if you use the same name for the HTML element name attributes and the variables in the state object you can do away with one of these functions, you do not need one function per DOM element than changes state. 

Example, input name='title' 
select name = 'category'  
just thesame as you have it in state. 
Then you can destructure the name from event.target and use it to setstate